subsys/nrf_security/src/core/lite/psa_core_lite.c

```diff
@@ -0,0 +1,351 @@
+	defined(PSA_WANT_ALG_PURE_EDDSA)
+
+psa_status_t psa_verify_message(
+	mbedtls_svc_key_id_t key_id, psa_algorithm_t alg,
+	const uint8_t *input, size_t input_length,
+	const uint8_t *signature, size_t signature_length)
+{
+	psa_status_t status = PSA_ERROR_NOT_SUPPORTED;
+	psa_key_attributes_t attr;
+	uint8_t pub_key[pub_key_max_size];
+	size_t pub_key_size = 0;
+
+	if (input == NULL || signature == NULL) {
+		return PSA_ERROR_INVALID_ARGUMENT;
+	}
+
+	if (!UTIL_CONCAT_OR(
+		VERIFY_ALG_ED25519(alg),
+		VERIFY_ALG_ECDSA_SECP_R1_256(alg),
+		VERIFY_ALG_DETERMINISTIC_ECDSA_SECP_R1_256(alg))) {
+		return PSA_ERROR_NOT_SUPPORTED;
+	}
+
+	status = get_key_buffer(key_id, &attr, pub_key, pub_key_max_size, &pub_key_size);
+	if (status != PSA_SUCCESS) {
+		return status;
+	}
+
+	return psa_driver_wrapper_verify_message(&attr, pub_key, pub_key_size,
+						 alg, input, input_length,
+						 signature, signature_length);
+}
+
+#endif
+
+/* Hash algorithms */
+
+#if defined(PSA_WANT_ALG_SHA_256) || defined(PSA_WANT_ALG_SHA_512)
+
+psa_status_t psa_hash_compute(
+	psa_algorithm_t alg, const uint8_t *input, size_t input_length,
+	uint8_t *hash, size_t hash_size, size_t *hash_length)
+{
+	if (input == NULL || hash_length == NULL) {
+		return PSA_ERROR_INVALID_ARGUMENT;
+	}
+
+	if (!UTIL_CONCAT_OR(VERIFY_ALG_SHA_256(alg),
+			    VERIFY_ALG_SHA_512(alg))) {
+		return PSA_ERROR_NOT_SUPPORTED;
+	}
+
+	return psa_driver_wrapper_hash_compute(alg, input, input_length,
+					       hash, hash_size, hash_length);
+}
+
+#endif /* PSA_WANT_ALG_SHA_256 || PSA_WANT_ALG_SHA_512 */
+
+/* Key establishment algorithms */
+
+/* Encryption algorithms */
+
+#if defined(PSA_WANT_ALG_CTR)
+
+/* Ensure that the largest key-size is supported */
+#if PSA_WANT_AES_KEY_SIZE_256
+#define ENC_KEY_MAX_SIZE	(32)
+#elif PSA_WANT_AES_KEY_SIZE_128
+#define ENC_KEY_MAX_SIZE	(16)
+#else
+#error "FW ecryption requires either AES-256 or AES-128 being enabled"
+#endif
+
+/** @brief Buffer containing the established/derived encryption key */
+uint8_t g_enc_key[ENC_KEY_MAX_SIZE];
+
+static psa_status_t get_enc_key(
+	mbedtls_svc_key_id_t key_id, psa_algorithm_t alg,
+	psa_key_attributes_t *attributes, size_t *key_length)
+{
+	if (attributes == NULL) {
+		return PSA_ERROR_INVALID_ARGUMENT;
+	}
+
+	if (IS_ENABLED(PSA_WANT_ALG_CTR) && alg != PSA_ALG_CTR) {
+		return PSA_ERROR_NOT_SUPPORTED;
+	}
+
+	return get_key_buffer(key_id, attributes, g_enc_key, ENC_KEY_MAX_SIZE, key_length);
+}
+
+static inline void clear_enc_key(void)
+{
+	safe_memzero(g_enc_key, ENC_KEY_MAX_SIZE);
+}
+
+
+psa_status_t psa_cipher_encrypt_setup(
+	psa_cipher_operation_t *operation, mbedtls_svc_key_id_t key, psa_algorithm_t alg)
+{
+	psa_status_t status = PSA_ERROR_NOT_SUPPORTED;
+	psa_key_attributes_t attributes;
+	size_t key_length;
+
+	if (operation == NULL) {
+		return PSA_ERROR_INVALID_ARGUMENT;
+	}
+
+	status = get_enc_key(key, alg, &attributes, &key_length);
+	if (status != PSA_SUCCESS) {
+		clear_enc_key();
```

Comment:
return status here and in all in the if statements of this function.

Reply:
Fixed